packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts

```diff
@@ -170,6 +174,39 @@ export function createMikrORMEntity() {
       return
     }
 
+    /**
+     * Defining an id property
+     */
+    if (field.dataType.name === "id") {
+      const IdDecorator = field.dataType.options?.primaryKey
+        ? PrimaryKey({
+            columnType: "text",
+            type: "string",
+            nullable: field.nullable,
+          })
+        : Property({
+            columnType: "text",
+            type: "string",
+            nullable: field.nullable,
+          })
+
+      IdDecorator(MikroORMEntity.prototype, field.fieldName)
+
+      /**
+       * Hook to generate entity within the code
+       */
+      MikroORMEntity.prototype.generateId = function () {
+        this.id = generateEntityId(this.id, "acc")
```

Comment:
todo: the prefix should be configuration, also, if not provided maybe we can infer a default value that depends on the model name as well wdyt?